src/aztec/Subsidy.sol

```diff
@@ -242,26 +222,17 @@ contract Subsidy is ISubsidy {
             return 0;
         }
 
-        uint256 subsidyAmount;
-
-        unchecked {
-            uint256 dt = block.timestamp - sub.lastUpdated; // sub.lastUpdated only update to block.timestamp, so can never be in the future
-
-            // time need to have passed for an incentive to have accrued
-            if (dt > 0) {
-                // The following should never overflow due to constraints on values. But if they do, it is bounded by the available subsidy
-                uint256 gasToCover = (dt * sub.gasPerMinute) / 1 minutes;
-                uint256 ethToCover = (gasToCover < sub.gasUsage ? gasToCover : sub.gasUsage) * block.basefee; // At maximum `sub.gasUsage` gas covered
+        uint256 subsidyAmount = _computeAccumulatedSubsidyAmount(sub);
 
-                subsidyAmount = ethToCover < sub.available ? ethToCover : sub.available;
-
-                sub.available -= uint128(subsidyAmount); // safe as `subsidyAmount` is bounded by `sub.available`
-                sub.lastUpdated = uint32(block.timestamp);
-
-                // Update storage
-                beneficiaries[_beneficiary].claimable = beneficiary.claimable + uint248(subsidyAmount); // safe as available is limited to u128
-                subsidies[msg.sender][_criteria] = sub;
+        if (subsidyAmount > 0) {
+            sub.lastUpdated = uint32(block.timestamp);
```

Comment:
Add a comment to the update of `lastUpdated`, similar to other other, defining why it is safe only to update when value is claimed. 

Safe as the three cases where subsidyAmount == 0, would be i), if there was a claim in the same block (time would be same, no need to update), ii) there are no available assets, which would have already returned 0 or iii) the amount is so small it is rounded down where we let the accumulated pass on to next claim.

Because the timestamp is reset to `block.timestamp` when subsidizing, a claim will not be getting "more" than planned from a stale timestamp.

Reply:
Addressed in e183db78c24b4fc5b3795e723f690248b8ed3940.